accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicMonotonicAccessioningWithAlternateRangesTest.java

```diff
@@ -0,0 +1,179 @@
+import uk.ac.ebi.ampt2d.test.service.TestMonotonicDatabaseService;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@ContextConfiguration(classes = {TestMonotonicDatabaseServiceTestConfiguration.class})
+public class BasicMonotonicAccessioningWithAlternateRangesTest {
+
+    private static final String CATEGORY_ID = "eva";
+    private static final String CATEGORY_ID_2 = "eva_2";
+    private static final String UNKNOWN_CATEGORY = "Unknown-Category";
+    private static final String INVALID_BLOCK_SIZE_CATEGORY = "invalid-block-size-test";
+    private static final String INVALID_BLOCK_START_VALUE_CATEGORY ="invalid-block-start-value-test";
+    private static final String INVALID_NEXT_BLOCK_INTERVAL_CATEGORY="invalid-next-block-interval-test";
+    private static final String INSTANCE_ID = "test-instance";
+    private static final String INSTANCE_ID_2 = "test-instance_2";
+
+    @Autowired
+    private TestMonotonicDatabaseService databaseService;
+    @Autowired
+    private ContiguousIdBlockService contiguousIdBlockService;
+
+    @Test(expected = BlockInitializationException.class)
+    public void testUnknownCategory() throws AccessionCouldNotBeGeneratedException {
+        List<AccessionWrapper<TestModel, String, Long>> evaAccessions =
+                getAccessioningService1(UNKNOWN_CATEGORY, INSTANCE_ID)
+                        .getOrCreate(getObjectsForAccessionsInRange(1, 10));
+    }
+
+    @Test(expected = BlockInitializationException.class)
+    public void testInvalidBlockSize() throws AccessionCouldNotBeGeneratedException {
+        List<AccessionWrapper<TestModel, String, Long>> evaAccessions =
+                getAccessioningService1(INVALID_BLOCK_SIZE_CATEGORY, INSTANCE_ID)
+                        .getOrCreate(getObjectsForAccessionsInRange(1, 10));
+    }
+
+    @Test(expected = BlockInitializationException.class)
+    public void testInvalidBlockStartValue() throws AccessionCouldNotBeGeneratedException {
+        List<AccessionWrapper<TestModel, String, Long>> evaAccessions =
+                getAccessioningService1(INVALID_BLOCK_START_VALUE_CATEGORY, INSTANCE_ID)
+                        .getOrCreate(getObjectsForAccessionsInRange(1, 10));
+    }
+
+    @Test(expected = BlockInitializationException.class)
+    public void testInvalidNextblockInterval() throws AccessionCouldNotBeGeneratedException {
+        List<AccessionWrapper<TestModel, String, Long>> evaAccessions =
+                getAccessioningService1(INVALID_NEXT_BLOCK_INTERVAL_CATEGORY, INSTANCE_ID)
+                        .getOrCreate(getObjectsForAccessionsInRange(1, 10));
+    }
+
+    @Test
+    public void testAlternateRangesWithPrefixes() throws AccessionCouldNotBeGeneratedException {
```

Comment:
This method is the only one that uses directly a `MonotonicAccessionGenerator`, so either we change to use the service or we move it to the appropriate test class.

Reply:
moving it to decoratedAccessionGenerator calss